Add section on website modes (#125)
epiverse-trace/blueprints: website.qmd

### Package website modes

Epiverse-TRACE packages use the [`{pkgdown}` package](https://pkgdown.r-lib.org/) to generate websites. `{pkgdown}` provides [development modes](https://pkgdown.r-lib.org/reference/build_site.html#development-mode) (specified by a package's [`_pkgdown.yml` file](https://pkgdown.r-lib.org/articles/customise.html)) to generate different sites for the development and released versions of your package.

When the package is not yet on CRAN, we use the `unreleased` development mode. In this case, we use a visual element to highlight the package development version in red. An example is the [website for `{superspreading}`](https://epiverse-trace.github.io/superspreading/) and its associated `_pkgdown` [configuration](https://github.com/epiverse-trace/superspreading/blob/main/_pkgdown.yml#L12-L13).

> After the package is accepted on CRAN, we switch to `auto` mode, which provides two versions of the website: `release` and `devel`. Here, an example is the `{cfr}`package [website](https://epiverse-trace.github.io/cfr/), which is configured [here](https://github.com/epiverse-trace/cfr/blob/main/_pkgdown.yml#L5-L6). We use this configuration for the CRAN packages because we expect most users to install the package from CRAN and so we direct them to the CRAN site associated with the current release. The additional `devel` option here allows users to manually switch to it if they installed the development version from GitHub.

Author: ```suggestion
After the package is accepted on CRAN, we switch to `auto` mode, which provides two versions of the website: `release` and `devel`. Here, an example is the `{cfr}`package [website](https://epiverse-trace.github.io/cfr/), which is configured [here](https://github.com/epiverse-trace/cfr/blob/main/_pkgdown.yml#L5-L6). We use this configuration for the CRAN packages because we expect most users to install the package from CRAN and so we direct them to the CRAN site associated with the current release. The additional `devel` option here allows users to manually switch to it (by adding `dev/` to the end of the website URL) if they installed the development version from GitHub. This version of the website will contain updates that are yet to be pushed to the CRAN version.
```

### Package website modes

Epiverse-TRACE packages use the [`{pkgdown}` package](https://pkgdown.r-lib.org/) to generate websites. `{pkgdown}` provides [development modes](https://pkgdown.r-lib.org/reference/build_site.html#development-mode) (in the [`_pkgdown.yml` file](https://pkgdown.r-lib.org/articles/customise.html)) to generate different sites for the development and released versions of your package.

> When the package is not yet on CRAN, we use the `unreleased` development mode. In this case, we use a visual element to highlight the package development version in red. An example is the [website](https://epiverse-trace.github.io/superspreading/) for `{superspreading}` and its associated `_pkgdown` [configuration](https://github.com/epiverse-trace/superspreading/blob/main/_pkgdown.yml#L12-L13).

Reviewer: Could expand the hyperlink to include "website for `{superspreading}`, but it's personal preference.

Author: I've done that in [677d2aa](https://github.com/epiverse-trace/blueprints/pull/125/commits/677d2aadad9701379c257f8d9437d30c29456dae).

> When the package is not yet on CRAN, we use the `unreleased` development mode. In this case, we use a visual element to highlight the package development version in red. An example is the [website](https://epiverse-trace.github.io/superspreading/) for `{superspreading}` and its associated `_pkgdown` [configuration](https://github.com/epiverse-trace/superspreading/blob/main/_pkgdown.yml#L12-L13).

Author: ```suggestion
When the package is not yet on CRAN, we use the `unreleased` development mode. In this case, we use a visual element to highlight the package development version in red. An example is the [website for `{superspreading}`](https://epiverse-trace.github.io/superspreading/) and its associated `_pkgdown` [configuration](https://github.com/epiverse-trace/superspreading/blob/main/_pkgdown.yml#L12-L13).
```

> After the package is accepted on CRAN, we switch to `auto` mode, which provides two versions of the website: `release` and `devel`. Here, an example is the `{cfr}`package [website](https://epiverse-trace.github.io/cfr/), which is configured [here](https://github.com/epiverse-trace/cfr/blob/main/_pkgdown.yml#L5-L6). We use this configuration for the CRAN packages because we expect most users to install the package from CRAN and so we direct them to the CRAN site associate with the installed release. The additional `devel` option here allows users to manually switch to it if they installed the development version from GitHub.

Reviewer: ```suggestion
After the package is accepted on CRAN, we switch to `auto` mode, which provides two versions of the website: `release` and `devel`. Here, an example is the `{cfr}`package [website](https://epiverse-trace.github.io/cfr/), which is configured [here](https://github.com/epiverse-trace/cfr/blob/main/_pkgdown.yml#L5-L6). We use this configuration for the CRAN packages because we expect most users to install the package from CRAN and so we direct them to the CRAN site associated with the current release. The additional `devel` option here allows users to manually switch to it if they installed the development version from GitHub.
```

> After the package is accepted on CRAN, we switch to `auto` mode, which provides two versions of the website: `release` and `devel`. Here, an example is the `{cfr}`package [website](https://epiverse-trace.github.io/cfr/), which is configured [here](https://github.com/epiverse-trace/cfr/blob/main/_pkgdown.yml#L5-L6). We use this configuration for the CRAN packages because we expect most users to install the package from CRAN and so we direct them to the CRAN site associate with the installed release. The additional `devel` option here allows users to manually switch to it if they installed the development version from GitHub.

Reviewer: > The additional `devel` option here allows users to manually switch to it if they installed the development version from GitHub.

This sentence is unclear to me. I feel it would be more informative to fully explain that the default site under `auto` is `release` but there is a `devel` site also rendered and give the users instructions on how to access the `devel` site. From what I can see on the Epiverse packages that use `auto` there is no direct link to `devel` sites so users need to type `/dev` into the URL, but there may be a better way to do this that I'm unaware of.

Author: Does 3d55eb92362c03f1308630dc51d784f196e76181 clarify it?

Reviewer: Yes! 👍

> We also provide text in package READMEs to direct users to the appropriate site. Typical text for the README might be: "The package has two websites: one for `[the stable release version on CRAN](URL)`, and another for `[the version in development](URL)`", where `URL` is the link to the respective site. See an example in the [`{EpiNow2}`] package [README](https://github.com/epiforecasts/EpiNow2/blob/main/README.Rmd#L125-L127).

Reviewer: I would reword this from "We also provide" to something like "It is recommended to provide", as {EpiNow2} is not an Epiverse hosted package and I'm unaware of any Epiverse packages that state this in their READMEs (please point out packages that do this as it would be good to know).

Author: There's none but we need to start doing that so that users are not confused when they visit the website and it doesn't correspond to the CRAN version. I'll remove it and we can add it after a wider discussion.

> We also provide text in package READMEs to direct users to the appropriate site. Typical text for the README might be: "The package has two websites: one for `[the stable release version on CRAN](URL)`, and another for `[the version in development](URL)`", where `URL` is the link to the respective site. See an example in the [`{EpiNow2}`] package [README](https://github.com/epiforecasts/EpiNow2/blob/main/README.Rmd#L125-L127).

Author: ```suggestion
```
